pydantic/env_settings.py

```diff
@@ -134,30 +136,47 @@ def __repr__(self) -> str:
 class EnvSettingsSource:
     __slots__ = ('env_file', 'env_file_encoding')
 
-    def __init__(self, env_file: Optional[StrPath], env_file_encoding: Optional[str]):
-        self.env_file: Optional[StrPath] = env_file
+    def __init__(self, env_file: Optional[DotenvType], env_file_encoding: Optional[str]):
+        self.env_file: Optional[DotenvType] = env_file
         self.env_file_encoding: Optional[str] = env_file_encoding
 
+    @staticmethod
+    def _read_process_env_vars(case_sensitive: bool) -> MutableMapping[str, str]:
+        if case_sensitive:
+            return os.environ
+        else:
+            return {k.lower(): v for k, v in os.environ.items()}
+
+    @staticmethod
+    def _read_dotenv_vars(
+        env_files: Optional[DotenvType], env_file_encoding: Optional[str], case_sensitive: bool
+    ) -> Dict[str, Optional[str]]:
+        def read_dotenv_file(env_file: StrPath) -> Dict[str, Optional[str]]:
```

Comment:
instead of implementing a method like, this - surely simpler to do something like

```py
if isinstance(env_files, (str, os.PathLike)):
    env_files = [env_files]

env = {}
for path in env_files:
    ...
```